pokemongo_bot/__init__.py

```diff
@@ -78,15 +78,26 @@ def _setup_event_system(self):
         # self.event_manager.emit('location', 'level'='info', data={'lat': 1, 'lng':1}),
 
     def tick(self):
+        start_time = time.time()
+
         self.cell = self.get_meta_cell()
         self.tick_count += 1
 
         # Check if session token has expired
         self.check_session(self.position[0:2])
 
         for worker in self.workers:
-            if worker.work() == WorkerResult.RUNNING:
-                return
+            worker_start = time.time()
+            result = worker.work()
+            worker_time = time.time() - worker_start
+
+            if result == WorkerResult.RUNNING:
+                break
+
+        elapsed_time = time.time() - start_time
+        sleep_time = max(0, self.config.minimum_tick_time - elapsed_time)
+        if sleep_time > 0:
```

Comment:
yeah this, and some of the  `worker_start` and `worker_time` lines were actually part of a commit that involved performance logging.  I'll remove these.